### ISSUE TITLE
Provide a way to prohibit certain HTTP method

### ISSUE THREAD

**Issue by minwoox** [line/armeria#3114]
Certain HTTP method such as `TRACE` is not allowed to use in some company.
In order to disable it, a user needs to do:
```java
sb.route()
  .methods(HttpMethod.TRACE)
  .pathPrefix("/")
  .build((ctx, req) -> HttpResponse.of(HttpStatus.FORBIDDEN));
```
which is a little verbose.

**Comment by trustin** [comment]
How about making path-related properties optional for route? i.e.

```java
sb.routeDecorator()
  .methods(HttpMethod.TRACE)
  .build((delegate, ctx, req) -> HttpResponse.of(HttpStatus.FORBIDDEN));
```

Still a little bit verbose, though. What else do you have in mind, adding a dedicated shortcut build method like `disableMethods(...)` ?

**Comment by minwoox** [comment]
> adding a dedicated shortcut build method like disableMethods(...) ?

Yeah, I was sort of thinking that but I'm also afraid that `ServerBuilder` is getting bigger.

**Comment by jorgheymans** [comment]
> ```java
> sb.routeDecorator()
>   .methods(HttpMethod.TRACE)
>   .pathPrefix("/")
>   .build((delegate, ctx, req) -> HttpResponse.of(HttpStatus.FORBIDDEN));
> ```
> 
> which is a little verbose.

I agree it's little verbose but the syntax is very clear to me. 

A shortcut method would indeed be nice, maybe something like `disableMethods(HttpMethod[] methods, HttpStatus status)` so that we get to pick the status code to associate ? `405` could be default.
